core/src/executor/evaluate/mod.rs

```diff
@@ -19,11 +19,13 @@ use {
         stream::{self, StreamExt, TryStreamExt},
     },
     im_rc::HashMap,
-    std::{borrow::Cow, rc::Rc},
+    std::{borrow::Cow, collections::HashMap as StdHashMap, rc::Rc},
 };
 
 pub use {error::EvaluateError, evaluated::Evaluated, stateless::evaluate_stateless};
 
+use crate::translate::TranslateError;
```

Comment:
Here we are in `evaluate` module, so rather than using `TranslateError` we can simply use `EvaluateError`.
To decide what type of `Error` to use, we can simply choose based on location and name of the module.